vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionSequencesController.java

```diff
@@ -0,0 +1,44 @@
+package edu.asu.diging.vspace.web;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.ISequence;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+
+@Controller
+public class ExhibitionSequencesController {
+
+    @Autowired
+    private IModuleManager moduleManager;
+
+    @Autowired
+    private ISequenceManager sequenceManager;
+
+    @RequestMapping(value = "/exhibit/module/{moduleId}/sequence/{sequenceId}")
+    public String sequence(Model model, @PathVariable("sequenceId") String sequenceId,
+            @PathVariable("moduleId") String moduleId) {
+        IModule module = moduleManager.getModule(moduleId);
+        model.addAttribute("module", module);
+        List<ISequence> sequences = moduleManager.getModuleSequences(moduleId);
+        List<ISlide> slides = sequenceManager.getSequence(sequenceId).getSlides();
```

Comment:
what if sequence doesn't exist? this will throw an exception, won't it?

Reply:
In this case, control will end in ExhibitionModuleController and the user will be shown a message saying "This module is not yet configured".

Reply:
someone can still bookmark for example a url that goes directly to a sequence and then this will throw an exception.